### PR TITLE
Generalize Axes __repr__ to 3D

### PR DESCRIPTION
**Pull request by scottshambaugh**
## PR Summary
Axes3D did not have the zlabel included in its repr, this adds that. We could cut down on code duplication by modifying the base axes string, but that seems a little sketchy.

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [n/a] New features are documented, with examples if plot related.
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [n/a] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Review by oscargus — APPROVED** [review]
Makes sense. A test would be nice and useful. Especially if one is trying to merge things later.

**Comment by anntzer** [comment]
I would suggest just writing the base class `__repr__` as something like (untested)
```python
        for name, axis in self._axis_map:
            if axis.get_label():
                fields += [f"{name}label={axis.get_label().get_text()!r}"]
```

**Comment by scottshambaugh** [comment]
@anntzer that's way nicer, changed

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -732,10 +732,9 @@ def __repr__(self):
                     titles.append(f"{k!r}:{title!r}")
         if titles:
             fields += ["title={" + ",".join(titles) + "}"]
-        if self.get_xlabel():
-            fields += [f"xlabel={self.get_xlabel()!r}"]
-        if self.get_ylabel():
-            fields += [f"ylabel={self.get_ylabel()!r}"]
+        for name, axis in self._axis_map.items():
+            if axis.get_label():
```
Comment (oscargus) [inline lib/matplotlib/axes/_base.py:736]:
```suggestion
            if axis.get_label() and axis.get_label().get_text():
```
?

**Comment by scottshambaugh** [comment]
Tests added

**Review by QuLogic — APPROVED** [review]

**Pull request merged**